### ISSUE TITLE
OGR error: sqlite3_open(layer 1 densify.gpkg)

### ISSUE THREAD

**Issue by jan-bausch** [juliepierson/qgis_line_similarity#4]
Hey! :)

I'm getting this error when trying to compare two layer:

```
QGIS version: 3.30.2-'s-Hertogenbosch
QGIS code revision: 0992b53397
Qt version: 5.15.2
Python version: 3.9.5
GDAL version: 3.3.2
GEOS version: 3.9.1-CAPI-1.14.2
PROJ version: Rel. 8.1.1, September 1st, 2021
PDAL version: 2.3.0 (git-version: Release)
Algorithm started at: 2023-09-05T11:52:06
Algorithm 'Line similarity' starting…
Input parameters:
{ 'ID_INPUT1' : 'fid', 'ID_INPUT2' : 'fid', 'INPUT1' : '/Users/jan.bausch/Documents/I-Route-Cargobike/QGIS/Berlin_compare_lenght/sp_data_od_anonym_export.gpkg|layername=sp_data_od_anonym_export_berlin', 'INPUT2' : '/Users/jan.bausch/Documents/I-Route-Cargobike/QGIS/Berlin_compare_lenght/sp_data_od_anonym_export.gpkg|layername=sp_data_od_anonym_export_berlin', 'INTERVAL' : 0, 'OUTPUT_FOLDER' : 'TEMPORARY_OUTPUT' }

Could not create layer layer 1 densify.gpkg: Creation of data source failed (OGR error: sqlite3_open(layer 1 densify.gpkg) failed: unable to open database file)
Execution failed after 0.03 seconds

Loading resulting layers
Algorithm 'Line similarity' finished
```

Do you maybe have an idea, what I can do to fix it?

Thanks a lot,
Jan

**Comment by juliepierson** [comment]
Hi, thanks for your feedback, sorry for this error ;-)

Unfortunately I won't have much time to answer you before friday. But it looks like the plugin can't create the temporary layer "densify.gpkg". Perhaps [this post](https://gis.stackexchange.com/a/435213) can provide some answers ? Also the interval needs to be set to a value greater than zero (this being still experimental, there isn't any checking on the user input parameters).

If you send me a sample of your data I'll try it on friday !

**Comment by jan-bausch** [comment]
Thanks a lot for answering :-)

Increasing the interval and choosing different output folders sadly didn't solve the problem for me.

[Here](https://drive.google.com/file/d/1ZdktQTHkCy4HjKYVk1AMyTTSbGcf8jCg/view?usp=sharing) are the project files. I want to compare the layers `sp_data_od_anonym_export` and `directions` over the `FID` field.

**Comment by jan-bausch** [comment]
Could you take a look at the project files? Sorry to bother :D

**Comment by juliepierson** [comment]
Yes, sorry about the delay ! I just had time for a quick look, but I could get it to work by creating another id field based on the original fid field, for both layers, and then using that new id field in the line similarity tool. It seems that since the fid field is a special field in geopackages, using it causes an error, but I haven't had time to look into it deeper to understand exactly what is the problem and to fix it.

Could you try it and let me know if it works ? Thanks !

Another thing to watch is that your input layers are in different crs, but this doesn't seem to cause a problem.

**Comment by juliepierson** [comment]
Generally speaking, this tool is a bit buggy, I wrote it for a special use case and I haven't had much time to check it. Feel free to look into the code and make improvements if you feel like it  :-)

**Comment by jan-bausch** [comment]
<img width="1433" alt="image" src="https://github.com/juliepierson/qgis_line_similarity/assets/15345398/e3a1c374-548f-448a-9b42-edb89e2fa9a5">


I still get the same issue, even when duplicating the fid field. Did you do something else? I created it using the Layer Properties  -> Field Calculator

**Comment by jan-bausch** [comment]
Still couldn't get it to run :/ Could you take a look again?

**Comment by juliepierson** [comment]
Sorry, it works for me like this :
![image](https://github.com/juliepierson/qgis_line_similarity/assets/6247389/6484c8e8-d340-4e51-b3b0-aa0a957e174b)

Best practice is that both layers use the same CRS. If not, select a projected CRS for first layer so that the interval unit is meters and not degrees (see the warning sign next to the interval parameter in your previous screenshot).

Also, did you quit editing mode for both layers after adding new id field ?

You can also check that you have write access to the temp folder used by the processing toolbox (Settings menu > Options > Processing > General). But if this wasn't the case you probably would have had problems before using the processing toolbox.

Perhaps you already checked all this, but I can't think of anything else right now...

**Comment by jan-bausch** [comment]
<img width="1143" alt="image" src="https://github.com/juliepierson/qgis_line_similarity/assets/15345398/b7511f86-b8fc-4d9c-92f9-337b2535d3e9">

Mhm, I still can't get it to work. 🤔 

- I've added the duplicated id field (as Integer64 to be precise)
- I've quit editing mode
- I have write access to the temp folder

[Here](https://drive.google.com/file/d/1KxWOp11dPP7VbU0smgTIKaS_F6uZMKqm/view?usp=sharing) you can find my modified project files. Does it work for you with these files? Also, could you send me your modified files so I can test it locally as well?

Thanks again for all the help!

**Comment by jan-bausch** [comment]
I didn't change the projection though, but it worked for you anyways, didn't it?

**Comment by juliepierson** [comment]
Hi, the files are only available with a google account. Thanks !

**Comment by jan-bausch** [comment]
Oh, just fixed the link, thx

**Comment by juliepierson** [comment]
Hi, your files work ok for me. Did you try running just the "densify by interval" tool on your layers, in the processing toolbox ?

**Comment by jan-bausch** [comment]
Just tried, and they work without a problem for both layers 🤔

**Comment by jan-bausch** [comment]
I'm using QGIS 3.30.2 by the way

**Comment by jan-bausch** [comment]
The error is "Could not create layer layer 1 densify.gpkg: Creation of data source failed (OGR error: sqlite3_open(layer 1 densify.gpkg) failed: unable to open database file)" 

Could it have something to do with the filename? Maybe the spaces are a problem?

**Comment by jan-bausch** [comment]
My best guess is that the error is caused somewhere [here](https://github.com/juliepierson/qgis_line_similarity/blob/b7caaa40b7edd560e2dc382068efb590dfe472ab/line_similarity_algorithm.py#L233C5-L234)

**Comment by jan-bausch** [comment]
The "Points Along Geometry" algorithm has also been changed, according to [here](https://github.com/qgis/QGIS/issues/34022). So could also be a reason 🤔

**Comment by juliepierson** [comment]
Yes, good idea, did you try the points along geometry algorithm on both your layers ?

**Comment by juliepierson** [comment]
But I use QGIS 3.28 and it works ok for me ; the bug you mention seems related to older versions ?

**Comment by jan-bausch** [comment]
> Yes, good idea, did you try the points along geometry algorithm on both your layers ?

Just did, and it works for both layers 🤔 


> But I use QGIS 3.28 and it works ok for me ; the bug you mention seems related to older versions ?

Ok, good that you are using a newer version too!

**Comment by jan-bausch** [comment]
Could you maybe send me the resulting csv. That would already help me to unblock another person 👌

**Comment by juliepierson** [comment]
Hi, sorry about the delay, here are the CSVs :
[layer1.csv](https://github.com/juliepierson/qgis_line_similarity/files/12684537/layer1.csv)
[layer2.csv](https://github.com/juliepierson/qgis_line_similarity/files/12684540/layer2.csv)
[similarity.csv](https://github.com/juliepierson/qgis_line_similarity/files/12684541/similarity.csv)

I used these parameters :
![image](https://github.com/juliepierson/qgis_line_similarity/assets/6247389/3eded7dd-21c3-4be6-bb4c-8b0691ebd033)

Hope this helps, keep in mind this is experimental !
